hyperspy/signal.py

```diff
@@ -2073,9 +2073,9 @@ def _plot_x_results(factors, loadings, factors_navigator, loadings_navigator,
                     factors_dim, loadings_dim):
     factors.axes_manager._axes[0] = loadings.axes_manager._axes[0]
     if loadings.axes_manager.signal_dimension > 2:
-        loadings.axes_manager.signal_dimension = loadings_dim
+        loadings.axes_manager._set_signal_dimension(loadings_dim)
```

Comment:
a few codecov warnings

Reply:
Done. For the other one, `IntegrateArea` and `integrate_in_range` are deprecated and I don't think it is worth it, particularly since this is testing a UI function, for which writing a test is a bit of a pain!